### PR TITLE
Fix controller upgrade process crash

### PR DESCRIPTION
**Pull request by tsurai**
Fixes various edge cases that caused the recreation of the node container to fail after an image update

### CODE REVIEW

File: build/init_and_run.sh [hunk 1]
```diff
@@ -16,11 +16,13 @@ echo "INDY_NODE_IP=${INDY_NODE_IP:=0.0.0.0}"
 echo "INDY_NODE_PORT=${INDY_NODE_PORT:=9701}"
 echo "INDY_CLIENT_IP=${INDY_CLIENT_IP:=0.0.0.0}"
 echo "INDY_CLIENT_PORT=${INDY_CLIENT_PORT:=9702}"
+echo "CONTROLLER_CONTAINER_NAME=${CONTROLLER_CONTAINER_NAME:=0.0.0.0}"
```
Comment (Echsecutor) [inline build/init_and_run.sh:19]:
```suggestion
echo "CONTROLLER_CONTAINER_NAME=${CONTROLLER_CONTAINER_NAME:=indy_node_controller}"
```

File: run/etc_indy/indy_config.py [hunk 2]
```diff
@@ -31,3 +31,6 @@
 
 # Directory to store logs. You might want to mount this in order to access the log files from outside the container.
 LOG_DIR = '/var/log/indy'
+
+# Hostname of the controller 
+controlServiceHost='indy_node_controller'
```
Comment (Echsecutor) [inline run/etc_indy/indy_config.py:36]:
```suggestion

```

written/overriden by init script anyway

**Review by Echsecutor — APPROVED** [review]
@tsurai tested this ;)

**Comment by c2bo** [comment]
https://github.com/hyperledger/indy-node-container/pull/124#issuecomment-1424372944 should hopefully fix the problem with the tests

**Pull request merged**